test/built-ins/Temporal/PlainMonthDay/prototype/with/basic.js

```diff
@@ -38,3 +39,8 @@ assert.throws(TypeError, () => md.with({ months: 12 }), "with({months})");
 
 TemporalHelpers.assertPlainMonthDay(md.with({ monthCode: "M12", days: 1 }),
   "M12", 15, "with({monthCode, days})");
+
+if (TemporalHelpers.nonDefaultCalendarId) {
+  const calendarMonthDay = Temporal.PlainMonthDay.from({ year: 2021, month: 1, day: 15, calendar: TemporalHelpers.nonDefaultCalendarId });
+  assert.throws(TypeError, () => calendarMonthDay.with({ month: 12 }), "nonIso8601MonthDay.with({month})");
+}
```

Comment:
The expected failure is now dependent upon use of a non–ISO-8601 calendar.

Reply:
I'm not sure that this is a change we want to make. See https://github.com/tc39/proposal-temporal/pull/2663#issuecomment-1688877251

Reply:
Replied at https://github.com/tc39/proposal-temporal/pull/2663#issuecomment-1690539352 ; I think this PR should land even if a further followup changes it.